### ISSUE TITLE
Response of deleteAllQueueItems operation

### ISSUE THREAD

**Issue by mcweba** [swisspush/vertx-redisques#32]
## Problem
The _deleteAllQueueItems_ operation is used to delete all queue items of a single queue. Apparently there's a discrepancy between the EventBus API and the HTTP API.

The HTTP-API returns a http status code 200 (OK) for every _deleteAllQueueItems_ request, even if nothing was actually deleted.

The EventBus API returns a status **OK** only if at least 1 queue items where deleted. Otherwise a status "ERROR" is returned.

The behaviour of both APIs is not ideal in my opinion.

### HTTP API
- A http DELETE request actually deleting the resource will return a http status code _200 OK_
- A http DELETE request to an already deleted resource will return a http status code _410 Gone_ or _404 Not Found_
- A http DELETE request which actually fails (because of database error for example) will return a http status code _500 Internal Server Error_

**Why http status code 410 Gone?**
A client receiving a http status code _410 Gone_ for a delete request will known that the resource was already deleted before the delete request.

### EventBus API
An EventBus DELETE request will always return a status **OK** with an additional information about the amount of queue items deleted. 

Example:
```
{
    "status": "ok",
    "value": <long AmountOfDeletedItems>
}
```
An EventBus DELETE request which actually fails (because of database error for example) will return a status **ERROR**.

Example:
```
{
    "status": "error"
}
```

**Comment by lbovet** [comment]
Yes, makes sense. I don't think we really have to distinguish between 410 and 404, though.

**Comment by mcweba** [comment]
Ok, so I would go with a 404 then. Do we have to consider backward compatibility since we would not return _ERROR_ (EventBus) anymore when nothing was deleted?

**Comment by mcweba** [comment]
Just started with this issue and saw that we already set the _value_ property when deleting over the EventBus API. However, the "count" is always 1 because we delete just 1 key in redis (a list holding the queue items). I would suggest to completely remove the _value_ property because we have not planned to add it in the HTTP API and also we would have to make an additional call to redis to get the count of queue items before we call delete.

**Comment by lbovet** [comment]
Makes sense. Provided nobody is expecting this value (mini breaking change)

**Comment by mcweba** [comment]
Further analysis of the code revealed, that the _value_ property of the EventBus API is required. The HTTP API is based on the EventBus API and we need information on the EventBus response whether a queue was deleted or not.
Suggestion for the EventBus API response:
```
{
    "status": "ok/error",
    "value": <long number of deleted queues>
}
```
So the _value_ property will be **1** when the queue was deleted or **0** when no queue was found. In both cases, the _status_ will be _ok_.

**Comment by mcweba** [comment]
Released in [v2.5.8](https://github.com/swisspush/vertx-redisques/releases/tag/v2.5.8)
